pkg/servicemesh/federation/common/util.go

```diff
@@ -38,9 +38,23 @@ func DefaultFederationCARootResourceName(instance *v1.ServiceMeshPeer) string {
 	return fmt.Sprintf("%s-ca-root-cert", instance.Name)
 }
 
-// EndpointsForService returns the Endpoints for the named service.
-func EndpointsForService(client kube.Client, name, namespace string) (*corev1.Endpoints, error) {
-	return client.KubeInformer().Core().V1().Endpoints().Lister().Endpoints(namespace).Get(name)
+// EndpointSlicesForService returns the Endpoints for the named service.
+func EndpointSlicesForService(client kube.Client, name, namespace string) ([]*discoveryv1.EndpointSlice, error) {
+	endpointSlices, err := client.KubeInformer().Discovery().V1().EndpointSlices().Lister().EndpointSlices(namespace).
+		List(kubelabels.Set(map[string]string{discoveryv1.LabelServiceName: name}).AsSelector())
```

Comment:
This can be simplified a bit.

```suggestion
		List(kubelabels.Set{discoveryv1.LabelServiceName: name}.AsSelector())
```

Reply:
Thanks, done.